### PR TITLE
Adding Appcenter test step in azure pipeline for testing in TestCloud

### PR DESCRIPTION
**Pull request by vsafonkin**
This PR added step in azure-pipelines.yml for running "appcenter.sh" script, that start TestCloud tests via appcenter-cli.

### CODE REVIEW

File: CalSmokeApp/Gemfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 source 'https://rubygems.org'
 
 gem 'calabash-cucumber', :github => "calabash/calabash-ios", :branch => "develop"
-gem "cucumber", "2.99.0"
+gem "cucumber", "~>2.0"
```
Comment (ivan-nosar) [inline CalSmokeApp/Gemfile]: Why do we need to do this? in my understanding `cucumber` version should not be changed

File: CalSmokeApp/bin/appcenter.sh [hunk 2]
```diff
@@ -34,21 +34,10 @@ BUILD_RUN_LOOP: iff 1, then rebuild Calabash iOS gem before uploading.
...
-if [ ! -e "${CREDS}" ]; then
-  error "This script requires a ${CREDS} file"
-  error "Generating a template now:"
-  cat >${CREDS} <<EOF
-export APPCENTER_TOKEN=
-EOF
-  cat ${CREDS}
-  error "Update the file with your credentials and run again."
-  error "Bye."
-  exit 1
```
Comment (ivan-nosar) [inline CalSmokeApp/bin/appcenter.sh]: In my understanding we shouldn't disallow to run this script locally. I suggest to implement the following:
- Check for `.appcenter-credentials` file. If it exists - load credentials from it.
- If `.appcenter-credentials` does not exist (perhaps this script performs on CI) - take credentials from `find-keychain-credential.sh`

File: CalSmokeApp/bin/make/ipa-cal.sh [hunk 3]
```diff
@@ -133,7 +133,7 @@ fi
 cat >"${XTC_DIR}/Gemfile" <<EOF
 source "https://rubygems.org"
 
-gem "calabash-cucumber", "2.99.0"
```
Comment (ivan-nosar) [inline CalSmokeApp/bin/make/ipa-cal.sh]: The same as above: why do we need to do this?

Comment (vsafonkin) [reply CalSmokeApp/bin/make/ipa-cal.sh]: In this case I see this error:
`Could not find gem 'calabash-cucumber (= 2.99.0)' in any of the gem sources`

File: CalSmokeApp/bin/appcenter.sh (hunk 2 above)
Comment (vsafonkin) [reply CalSmokeApp/bin/appcenter.sh]: Added check file with credentials.

File: CalSmokeApp/bin/make/ipa-cal.sh (hunk 3 above)
Comment (vsafonkin) [reply CalSmokeApp/bin/make/ipa-cal.sh]: The latest version of `calabash-cucumber` gem is 0.21.x

File: CalSmokeApp/bin/make/ipa-cal.sh [hunk 4]
```diff
@@ -133,7 +133,7 @@ fi
...
+gem "calabash-cucumber"
 gem "json", "1.8.6"
```
Comment (jmoody) [inline CalSmokeApp/bin/make/ipa-cal.sh]: No.  This needs to be "calabash-cucumber", "2.99.0"

We can use cucumber 3.x locally, but in Test Cloud we must use 2.x

**Comment by jmoody** [comment]
Please review the failures in CI before asking for review.

The failure is because the Gemfile.lock includes a reference to bunder 1.17.3 and in the pipeline we are using 2.0.  

I think the solution is remove the Gemfile.lock from git index.

File: CalSmokeApp/bin/make/ipa-cal.sh (hunk 4 above)
Comment (vsafonkin) [reply CalSmokeApp/bin/make/ipa-cal.sh]: But the latest version of calabash-cucumber gem is 0.21.10
https://rubygems.org/gems/calabash-cucumber/versions/0.21.10

**Comment by ivan-nosar** [comment]
> Please review the failures in CI before asking for review.
> 
> The failure is because the Gemfile.lock includes a reference to bunder 1.17.3 and in the pipeline we are using 2.0.
> 
> I think the solution is remove the Gemfile.lock from git index.

Sure, thanks for that point. We will be more attentive in future.
`Gemfile.lock` is not in index actually. We upgraded `bundler` version to `2.0.2`

File: CalSmokeApp/bin/make/ipa-cal.sh (hunk 4 above)
Comment (jmoody) [reply CalSmokeApp/bin/make/ipa-cal.sh]: Apologies, I continue to make this mistake - confusing `calabash-cucumber` and `cucumber`.

**Pull request merged**